pkg/nuclide-diagnostics-store/lib/LinterAdapter.js

```diff
@@ -148,35 +148,49 @@ export class LinterAdapter {
   }
 
   async _runLint(editor: TextEditor): Promise<void> {
-    if (this._enabled) {
-      const result = await this._requestSerializer.run(this._provider.lint(editor));
-      if (result.status === 'success') {
-        const buffer = editor.getBuffer();
-        if (buffer.isDestroyed()) {
-          return;
-        }
-
-        if (this._provider.invalidateOnClose && !this._onDestroyDisposables.has(buffer)) {
-          const disposable = buffer.onDidDestroy(() => {
-            this._invalidateBuffer(buffer);
-            this._onDestroyDisposables.delete(buffer);
-            disposable.dispose();
-          });
-          this._onDestroyDisposables.set(buffer, disposable);
-        }
-
-        const linterMessages = result.result;
-        const diagnosticUpdate = linterMessagesToDiagnosticUpdate(
-          editor.getPath(),
-          linterMessages, this._provider.providerName || this._provider.name,
-        );
+    if (!this._enabled) {
+      return;
+    }
+
+    const maybe = this._provider.lint(editor);
+    if (!maybe) {
+      return;
+    }
+
+    const result = await this._requestSerializer.run(maybe);
+    if (result.status !== 'success') {
+      return;
+    }
+
+    const buffer = editor.getBuffer();
+    if (buffer.isDestroyed()) {
+      return;
+    }
+
+    if (this._provider.invalidateOnClose && !this._onDestroyDisposables.has(buffer)) {
```

Comment:
Tried to make as small changes as possible until the spec passed. The `linterMessages = result.result` row was placed below the block so kept it there, just in case. Do you want me to move it?

The code would look more symmetrical if the guard was moved above the block, so I don't mind moving it.